src/Saturn.LiveView/LiveView.fs

```diff
@@ -0,0 +1,192 @@
+﻿namespace Saturn
+
+open Channels
+open Microsoft.AspNetCore.Http
+open Microsoft.Extensions.DependencyInjection
+open System.Threading.Tasks
+open Giraffe.GiraffeViewEngine
+open Elmish
+open FSharp.Control.Tasks.V2
+
+module LiveComponenet =
+  type ILiveComponenet =
+    abstract member InternalChannel : IChannel with get
+
+  type LiveComponentMsg = {Event: string; ElementId: string; Data: string}
+  type internal ViewUpdateMsg = {ComponentId: string; Data: string}
+
+[<AutoOpen>]
+module LiveComponentBuilder =
+  open LiveComponenet
+
+  type LiveComponenetBuilderState<'State, 'Msg> = {
+    Join: (HttpContext -> ClientInfo -> Task<JoinResult>) option
+    Init: (HttpContext -> ClientInfo -> (Cmd<'Msg> -> unit) -> Task<'State * Cmd<'Msg>>) option
+    Update: (HttpContext -> ClientInfo -> 'Msg -> 'State -> Task<'State * Cmd<'Msg>>) option
+    View: (HttpContext -> ClientInfo -> 'State -> XmlNode) option
```

Comment:
Using the concrete type `XmlNode` ties this implementation of live components with the view engine of Giraffe. It won't be possible to use [Feliz.ViewEngine](https://github.com/dbrattli/Feliz.ViewEngine) which uses a different type called `ReactElement`. 

Would it be possible to abstract the view output into the type declaration? i.e.
```fs
type LiveComponenetBuilderState<'State, 'Msg, 'View> = {
   View: (HttpContext -> ClientInfo -> 'State -> 'View) option
}

```